src/AwsMetadata.js

```diff
@@ -53,6 +53,9 @@ export default class AwsMetadata {
       this.lookupMetadataKey(`network/interfaces/macs/${results.mac}/vpc-id`, (err, vpcId) => {
         results['vpc-id'] = vpcId;
         this.logger.debug('Found Instance AWS Metadata', results);
+        Object.keys(results)
+              .filter(k => results[k] === null || results[k] === undefined)
+              .forEach(k => delete results[k]);
```

Comment:
Great suggestion. I'll update the PR. As you may have guessed, javascript is not my native language, but since my favorite language is Scala, the suggested functional style is appealing.